jbpm-human-task/jbpm-human-task-core/src/main/java/org/jbpm/services/task/assignment/impl/AssignmentServiceImpl.java

```diff
@@ -84,10 +85,29 @@ public void onTaskDone(Task task, TaskContext context) {
             return;
         }
         
-        this.strategy.taskDone(task, context);
+        computeAssignementStrategyForTask(task).taskDone(task, context);
         logger.debug("Assignment strategy notified about task {} being done", task);
     }
 
+    private AssignmentStrategy computeAssignementStrategyForTask(Task task) {
+        if(task.getTaskData().getTaskInputVariables().containsKey(TASK_IMPLICIT_INPUT_VAR_ASSIGNEMENT)) {
+            String userTaskStrategy = (String) task.getTaskData().getTaskInputVariables().get(TASK_IMPLICIT_INPUT_VAR_ASSIGNEMENT);
+            return userTaskStrategy != null ? registry.getStrategy(userTaskStrategy) : new AssignmentStrategy() {
```

Comment:
when userTaskStrategy is not null, the attribute (**this.strategy**) is not updated with the value of _registry.getStrategy(userTaskStrategy)_, so it's a little bit confusing for tracing like in above line:
`logger.warn("Strategy {} did not return any assignment for task {}", strategy, task);`

It would be better if this.strategy is set up with the chosen strategy in this case, before returning the result.

Reply:
fixed.